### PR TITLE
fix: Docker image tag parsing when user specifies custom image

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Fixes the problem reported in Slack by @Jimvin 
> For info, the thing that triggered this error for us was using a custom image from a local docker repo which had the port number in the url. I am assuming that it tried to parse everything after the first : character using SemVer, but the error isn’t all that helpful in pointing out what was being parsed.
This seems to be a bug in the parsing of custom image locations with port numbers in the location.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: src/commons/product_image_selection.rs [hunk 1]
```diff
@@ -140,12 +137,49 @@ impl ProductImage {
     }
 }
 
+fn extract_image_tag(image: &str) -> &str {
```
[the remark this answers is not in the document]
Comment (Techassi) [reply src/commons/product_image_selection.rs:140]: Either what @lfrancke mentioned or use a proper `FromStr` impl for parsing the image tag from an image URL.

[the remark this answers is not in the document]
Comment (sbernauer) [reply src/commons/product_image_selection.rs:140]: I was searching for something like that. @nightkr are you aware of any such functionality?

[the remark this answers is not in the document]
Comment (sbernauer) [reply src/commons/product_image_selection.rs:140]: Found https://docs.rs/dockerfile-parser/latest/dockerfile_parser/struct.ImageRef.html
I would use that if no one objects ;)

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply src/commons/product_image_selection.rs]: Not lovin' this sentence from that functions doc :) 

>  This is not fallible, however malformed image strings may return unexpected results.

Then again, this is not strictly speaking specified in any formal way, and the users will notice that it fails when images don't pull. Without admission controllers we don't have any feedback loop anyway ..

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply src/commons/product_image_selection.rs]: I agree. Once we have admission controllers we could consider adding a regex as a quick sanity check I think ..

[the remark this answers is not in the document]
Comment (sbernauer) [reply src/commons/product_image_selection.rs]: Once you approve we can merge I guess :)

**Review by soenkeliebau — APPROVED** [review]
Looked at the code and tested the changes locally.
Works as intended. Not aure if this is the final form of image parsing, bit fixes an issue.

**Pull request merged**